packages/query-core/src/hydration.ts

```diff
@@ -178,9 +183,18 @@ export function hydrate(
     }
 
     if (promise) {
+      const deserialize =
+        client.getDefaultOptions().hydrate?.deserialize ?? defaultDeserialize
+
+      const initialPromise = (async () => {
+        const result = await promise
+        const deserialized = deserialize(result)
+        return deserialized
+      })()
```

Comment:
for some reason i don't understand 

```ts
const initialPromise = promise.then((result) => deserialize(result))
```

results in a `Missing queryFn` error but this IIFE approach does work

Reply:
I think we should try to figure out why that is. There shouldn't be any difference between `promise.then` and this code.

Can you add a test case and then we can maybe look into why this is failing

Reply:
where is this tested? e2e?

Reply:
not sure how accurate that is since there's no RSC transport there but can give it a go in a bit